src/main/kotlin/onm/builder/HouseBuilder.kt

```diff
@@ -1,43 +1,86 @@
 package onm.builder
 
-import onm.configuration.FurnitureType
-import onm.configuration.RoomType
+import onm.configuration.*
 import onm.events.EventHandler
+import onm.events.IEventHandler
+import onm.house.devices.AbstractDevice
 import onm.house.devices.Fridge
 import onm.house.devices.Oven
 import onm.house.devices.Washer
 import onm.house.furniture.Furniture
+import onm.house.places.Room
 import onm.house.places.RoomBuilder
 import java.util.*
 
+
 /**
  * Class representing framework builder for devices and rooms in a smart house.
  */
-class HouseBuilder {
+class HouseBuilder private constructor(
+        /**
+         * Primary house reference.
+         * */
+        val house: House,
+
+        /**
+         * Event handler which will be used for devices created with config.
+         * */
+        private val eventHandler: IEventHandler) {
+
+    companion object {
+        /**
+         * Builds empty house.
+         * */
+        fun createEmptyHouseBuilder(): HouseBuilder {
+            val house = House()
+            val eventHandler = EventHandler() //todo make it singleton
+            return HouseBuilder(house, eventHandler)
+        }
 
+        /**
+         * Builds house from given config class. This class should be parsed from JSON.
+         * */
+        fun createHouseBuilder(config: ConfigurationDataClass): HouseBuilder {
+            val house = House()
+            val eventHandler = EventHandler() //todo make event handler as singleton
 
-    /**
-     * Event handler dealing with all raised events from i.e., devices
-     */
-    val eventHandler = EventHandler() //TODO later add via constructor and implement wrapper
+            val houseBuilder = HouseBuilder(house, eventHandler)
 
-    /**
-     * House variable containing references to all rooms
-     */
-    val house = House() //TODO later add via constructor and implement wrapper
+            for (key in config.roomsAndDevices.keys) {
+                val room = houseBuilder.addOrGetRoom(key)
+
+                for (deviceConfig in config.roomsAndDevices[key]!!) {
+                    room.addDevice(createDevice(deviceConfig, eventHandler))
+                }
+            }
+
+            return houseBuilder
+        }
+
+
+        private fun createDevice(deviceConfig: DeviceConfig, eventHandler: IEventHandler): AbstractDevice {
```

Comment:
Function does not return anything. Or does it?

Reply:
when statement is returning device

Reply:
my bad, `return` in front of `when` is missing